### ISSUE TITLE
Ampache-docker behind nginx-proxy with ssl configuration.

### ISSUE THREAD

**Issue by no95typem** [ampache/ampache#2662]
Hello. I'm trying to setup ampache in the docker container with nginx-proxy and ssl at home LAN (because I would like to get working it with Nextcloud-docker). The problem is that I can't play/download any media from subsonic clients.
I don't know very well how it all works, so I can't solve my problem myself and I am asking for help with my setup.

When I try to play any song from my subsonic client on android phone I get this in the log:
```
2020-12-23T18:17:21+00:00 [no95typem] (subsonic_api.class) -> Stream proxy: http://192.168.0.2:443/ampache/play/index.php?ssid=084478aee2ed01be32775588f8af2146&type=song&oid=169&uid=2&client=android&player=api&name=Anubis%20-%20Eyes%20Of%20The%20Storm.mp3 
2020-12-23T18:17:51+00:00 [no95typem] (subsonic_api.class) -> Stream error: Failed to connect to 192.168.0.2 port 80: Connection timed out 
2020-12-23T18:18:11+00:00 [no95typem] (subsonic_api.class) -> Stream error: Failed to connect to 192.168.0.2 port 443: Connection timed out 
2020-12-23T18:18:43+00:00 [no95typem] (subsonic_api.class) -> Stream error: Failed to connect to 192.168.0.2 port 443: Connection timed out
```
When I try to download this url from a browser - the song is downloading successfully.

So, looking at the connection address 192.168.0.2, I understand that I configured something wrong, because the address of my device is 192.168.0.6. Why connection is needed at all, the link is not transferred to the device such as links to album art? I probably need to know the subsonic protocol to understand this.

So I just can't figure it out - too much knowledge is needed which I don't have. Therefore, I ask for help in setting up ampache with nextcloud with nginx-proxy in LAN, maybe even examples of configurations.

**Client type:**

* SubSonic

## Settings

My current config for nginx:
```
server {
    listen 80;
    listen [::]:80;
    return 301 https://192.168.0.2$request_uri;
}

server {
        listen 443 ssl;
        ssl_certificate certs/default.crt;
        ssl_certificate_key certs/default.key;

        # Redirects for DAV clients
        location = /.well-known/carddav {
            return 301 $scheme://$host/nextcloud/remote.php/dav;
        }

        location = /.well-known/caldav {
            return 301 $scheme://$host/nextcloud/remote.php/dav;
        }

        location /nextcloud {

            rewrite ^/nextcloud(.*) $1 break;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_redirect off;
            proxy_pass http://nextcloud.localhost;
        }
        
        location /ampache {
            rewrite ^/ampache(.*) $1 break;
            rewrite ^/play/ssid/(.*)/type/(.*)/oid/([0-9]+)/uid/([0-9]+)/client/(.*)/player/(.*)/name/(.*)$ /play/index.php?ssid=$1&type=$2&oid=$3&uid=$4&client=$5&player=$6&name=$7 last;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            #proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header X-Forwarded-Host $http_host;
            proxy_set_header Host $http_host;
            proxy_max_temp_file_size 0;
            proxy_redirect off;
            proxy_pass http://ampache.localhost;
        }
}
```

Key parameters of ampache.cfg.php (I experimented with them - really affect performance only web_path and local_web_path:

```
http_host = "192.168.0.2"
http_port = 443
web_path = "/ampache"
local_web_path = "http://192.168.0.2:443/ampache"
force_ssl = "true"
```

**Comment by cquike** [comment]
I have a similar setup and I think that the problem is on the local_web_path setting:
From, the config/ampache.cfg.php template:
```
; The local http url of your server.
; This is used to access the server from within the
; same host where ampache is running.
; For example, if the ampache server is not
; directly accessed via the public domain but via a reverse
; proxy, local_web_path would need to be changed
; to a localhost URL.
; If not set, retrieved automatically from server information.
; DEFAULT: none
```

I think that, in your case, setting it to:
`local_web_path = "http://localhost"`
would do the work (assuming the docker container is listening on port 80, of course). Maybe you have to append "/ampache" at the end, depending on how the web server inside the docker container is setup.

Hope it helps...

**Comment by no95typem** [comment]
@cquike Thank You for the answer, for a long time I could not check this - I tried the docker rootless. So far I decided to return to the usual rootfull docker (I did not understand how to set up a reverse proxy for the simplest configuration in rootless mode). 
But this is a departure from the topic. I tried `local_web_path = "http://localhost"`, but the problem is still exists.
Meanwhile I also launched D-Sub on my android and it works, unlike other subsonic clients!
Now I am satisfied, but I would not refuse to find out why this is happening with other clients, and perhaps it can be useful, because the next android update can make D-Sub broken.
In addition, I also noticed that most of the covers are not loaded in normal quality. I will think about this later.
So issue may be closed. I think if there was someone here with suitable configs they would have already shared.

**Comment by no95typem** [comment]
I was wrong yesterday. I did not restart the container when I changed the config:
`local_web_path = "http://localhost"`
Today, after rebooting the PC, ampache stopped working even in the web interface. So I returned everything as it was:
`local_web_path = "http://192.168.0.2:443/ampache"`
And then I restarted the container. It worked again.
If I find anything else interesting for such configuration, I will share the configs. Perhaps someone it will be useful.
